polygon/bor/bordb/prune.go

```diff
@@ -88,6 +115,9 @@ func PruneBorBlocks(tx kv.RwTx, blockTo uint64, blocksDeleteLimit int, SpanIdAt
 		deleted++
 		counter--
 	}
+	if err != nil {
+		return deleted, err
+	}
 
 	if snaptype.CheckpointsEnabled() {
 		checkpointCursor, err := tx.RwCursor(kv.BorCheckpoints)
```

Comment:
I know this isn't part or purpose of the PR, but since this function is a bit long, you could split it into smaller sub-functions each deleting from a certain table.